cmd/bnscli/cmd_view.go

```diff
@@ -42,6 +43,38 @@ kind of operation are you authorizing.
 	if err != nil {
 		return fmt.Errorf("cannot JSON serialize: %s", err)
 	}
-	_, err = output.Write(pretty)
-	return err
+	_, _ = output.Write(pretty)
+
+	// When printing a transaction of a proposal, the embeded in proposal
+	// message is obfuscated. Extract it and print additionally.
+	_ = printProposalMsg(output, tx)
+
+	return nil
+}
+
+func printProposalMsg(output io.Writer, tx app.Tx) error {
+	msg, err := tx.GetMsg()
+	if err != nil {
+		return fmt.Errorf("cannot get transaction message: %s", err)
+	}
+	proposalMsg, ok := msg.(*gov.CreateProposalMsg)
+	if !ok {
+		return nil
+	}
+
+	var propTx app.Tx
```

Comment:
I think you should use app.ProposalOptions here:
https://github.com/iov-one/weave/blob/e8564ea251ff6b36d0f27f87d20dc81ae352ab6d/cmd/bnsd/app/codec.proto#L100-L117

This is what actually does the parsing in the governance package (for bnsd). And this (private function) show how you can use that to decode:
https://github.com/iov-one/weave/blob/e8564ea251ff6b36d0f27f87d20dc81ae352ab6d/cmd/bnsd/app/gov_handler.go#L16-L24

Reply:
Good catch!

I was a bit suspicious why it works, but never took time to investigate.